repos/system_upgrade/el7toel8/libraries/isccfg.py

```diff
@@ -0,0 +1,959 @@
+#!/usr/bin/env python
+#
+# Simplified parsing of bind configuration, with include support and nested sections.
+
+import re
+import string
+
+
+class ConfigParseError(Exception):
+    """ Generic error when parsing config file """
+
+    def __init__(self, message, error=None):
+        super(ConfigParseError, self).__init__(message)
+        self.error = error
+    pass
+
+
+class ConfigFile(object):
+    """ Representation of single configuration file and its contents """
+    def __init__(self, path):
+        """
+        Load config file contents from path
+
+        :param path: Path to file
+        """
+        self.path = path
+        self.load(path)
+        self.status = None
+
+    def __str__(self):
+        return self.buffer
+
+    def __repr__(self):
+        return 'ConfigFile {0} ({1})'.format(
+                self.path, self.buffer)
+
+    def load(self, path):
+        with open(path, 'r') as f:
+            self.buffer = self.original = f.read()
+            f.close()
```

Comment:
with open context closes the file for you, no need to do it manually

Reply:
ok